fs/nfs/rpc_clnt.c

```diff
@@ -371,13 +395,22 @@ static int rpcclnt_receive(FAR struct rpcclnt *rpc,
       resplen = mark;
     }
 
-  error = psock_recv(&rpc->rc_so, reply, resplen, 0);
+repeat:
+  error = psock_recv(&rpc->rc_so, reply + offset, resplen, 0);
   if (error < 0)
     {
       ferr("ERROR: psock_recv response failed: %d\n", error);
       return error;
     }
 
+  resplen -= error;
+  offset  += error;
+
+  if (rpc->rc_sotype == SOCK_STREAM && resplen)
+    {
+      goto repeat;
+    }
```

Comment:
According to the coding stardard, goto should only be used for exception handling:
https://cwiki.apache.org/confluence/display/NUTTX/Coding+Standard#goto

```suggestion
  do {
    error = psock_recv(&rpc->rc_so, reply + offset, resplen, 0);
    if (error < 0)
      {
        ferr("ERROR: psock_recv response failed: %d\n", error);
        return error;
      }

    resplen -= error;
    offset  += error;
  } while (rpc->rc_sotype == SOCK_STREAM && resplen != 0);
```

Reply:
@gustavonihei 

Thanks for the suggestion.
I will change the code based on the https://cwiki.apache.org/confluence/display/NUTTX/Coding+Standard#dowhile

Reply:
Done

Reply:
> @gustavonihei
> 
> Thanks for the suggestion.
> I will change the code based on the https://cwiki.apache.org/confluence/display/NUTTX/Coding+Standard#dowhile

@masayuki2009 I didn't notice I gave a non-compliant suggestion! :rofl:
Thanks for pointing out!